src/UserRepository.js

```diff
@@ -7,77 +7,89 @@ class UserRepository {
   }
 
   calculateCommunityAvgStepGoal() {
-    let communityStepGoals = this.users.map(user => user.dailyStepGoal);
+    let communityStepGoals = this.users.map((user) => user.dailyStepGoal);
     let communityTotal = communityStepGoals.reduce((sum, goal) => {
       sum += goal;
       return sum;
     }, 0);
     return communityTotal / this.users.length;
   }
 
-  // calculateAverageSleepQuality() {
-  //   let totalSleepQuality = this.users.reduce((sum, user) => {
-  //     sum += user.sleepQualityAverage;
-  //     return sum;
-  //   }, 0);
-  //   return totalSleepQuality / this.users.length;
-  // }
+  calculateAverageSleepQuality() {
+    let totalSleepQuality = this.users.reduce((sum, user) => {
+      sum += user.sleepQualityAverage;
+      return sum;
+    }, 0);
+    return totalSleepQuality / this.users.length;
+  }
 
-  // calculateAverageSteps(date) {
-  //   let allUsersStepsCount = this.users.map(user => {
-  //     return user.activityRecord.filter(activity => {
-  //       return activity.date === date;
-  //     });
-  //   })
-  //   let sumOfSteps = allUsersStepsCount
-  //     .reduce((stepsSum, activityCollection) => {
-  //       activityCollection.forEach(activity => {
-  //         stepsSum += activity.steps
-  //       })
-  //       return stepsSum;
-  //     }, 0);
-  //   return Math.round(sumOfSteps / allUsersStepsCount.length);
-  // }
+  findBestSleepers(date) {
+    return this.users.filter(user => user.calculateAverageQualityThisWeek(date) > 3);
+  }
 
-  // calculateAverageStairs(date) {
-  //   let allUsersStairsCount = this.users.map(user => {
-  //     return user.activityRecord.filter(activity => {
-  //       return activity.date === date;
-  //     });
-  //   })
+  getLongestSleepers(date, sleepRepository) {
+    let allSleepsOnDate = sleepRepository.filter((sleep) => sleep.date === date);
+    return allSleepsOnDate ? allSleepsOnDate.sort((a, b) => b.hoursSlept - a.hoursSlept).shift().userID : "Data not found";
+  }
 
-  //   let sumOfStairs = allUsersStairsCount.reduce((stairsSum, activityCollection) => {
-  //     activityCollection.forEach(activity => {
-  //       stairsSum += activity.flightsOfStairs
-  //     })
-  //     return stairsSum;
-  //   }, 0);
-  //   return Math.round(sumOfStairs / allUsersStairsCount.length);
-  // }
+  getWorstSleepers(date, sleepRepository) {
+    let allSleepsOnDate = sleepRepository.filter((sleep) => sleep.date === date);
+    return allSleepsOnDate ? allSleepsOnDate.sort((a, b) =>  a.hoursSlept - b.hoursSlept).shift().userID : "Data not found";
+  }
 
-  // calculateAverageMinutesActive(date) {
-  //   let allUsersMinutesActiveCount = this.users.map(user => {
-  //     return user.activityRecord.filter(activity => {
-  //       return activity.date === date;
-  //     });
-  //   })
+  calculateAverageSteps(date) {
+    let allUsersStepsCount = this.users.map(user => {
+      return user.activityRecord.filter(activity => {
```

Comment:
I hear tales that map and filter when used together can be refactored into reduce, so this may be a good opportunity to do so.